divi/src/test/kernel_tests.cpp

```diff
@@ -83,18 +83,14 @@ class CheckCoinstakeForVaultsTestFixture
 
 BOOST_FIXTURE_TEST_SUITE(CheckCoinstakeForVaults_tests, CheckCoinstakeForVaultsTestFixture)
 
-BOOST_AUTO_TEST_CASE(willFailNonCoinstakeTransactions)
+BOOST_AUTO_TEST_CASE(willAllowNonCoinstakeTransactions)
 {
   CMutableTransaction mtx;
   mtx.vin.push_back(CTxIn(nonVaultCoins[0]));
   mtx.vout.push_back(CTxOut(1, scriptNonVault));
   mtx.vout.push_back(CTxOut(15000 * COIN + CENT, scriptNonVault));
 
   BOOST_CHECK(!CTransaction(mtx).IsCoinStake());
-  BOOST_CHECK(!RunCheck(mtx));
-
-  mtx.vout[0].SetEmpty();
-  BOOST_CHECK(CTransaction(mtx).IsCoinStake());
   BOOST_CHECK(RunCheck(mtx));
 }
```

Comment:
We dont want to change the tests to fit the behaviour, but rather the other way around. This is probably best left unchanged since we want the check to explicitly disallow non-coinstakes.

Reply:
See other comment.  I changed the function because I think it makes no sense to explicitly check for coinstakes before calling it and then having a redundant check against coinstakes in the function.  Instead the function now really does what it should, namely "apply extra checks for coinstakes".

Reply:
Ok. Then maybe we can rename the test, in addition to the proposed changes to something like, willIgnoreNonCoinstakeTransactions

Reply:
I've renamed the test as suggested.  I also extended it to explicitly check that even if a transaction is non-valid as a coinstake (because it violates the rules), it will be accepted as a non-coinstake.